packages/matplotlib/test_matplotlib.py

```diff
@@ -69,74 +33,103 @@ def save_canvas_data(selenium, output_path):
     output_path.write_bytes(canvas_png)
 
 
-def compare_with_reference_image(selenium, reference_image):
-    reference_image_encoded = base64.b64encode(reference_image.read_bytes())
-    deviation = selenium.run(
-        f"""
-        import io
-        import base64
-        import numpy as np
-        from PIL import Image
-        canvas_data = plt.gcf().canvas.get_pixel_data()
-        ref_data = np.asarray(Image.open(io.BytesIO(base64.b64decode({reference_image_encoded!r}))))
-
-        deviation = np.mean(np.abs(canvas_data - ref_data))
-        float(deviation)
-        """
+@run_in_pyodide(packages=["matplotlib"])
+def patch_font_loading_and_dpi(selenium, handle, target_font=""):
+    """Monkey-patches font loading and dpi to allow testing"""
+    from matplotlib_pyodide.html5_canvas_backend import (
+        FigureCanvasHTMLCanvas,
+        RendererHTMLCanvas,
     )
 
-    # Note: uncomment this line if you want to save the output canvas image (for comparison).
-    # save_canvas_data(selenium, reference_image.with_name(f"output-{reference_image.name}"))
+    FigureCanvasHTMLCanvas.get_dpi_ratio = lambda self, context: 2.0
+    load_font_into_web = RendererHTMLCanvas.load_font_into_web
+
+    def load_font_into_web_wrapper(
+        self, loaded_font, font_url, orig_function=load_font_into_web
+    ):
+        fontface = orig_function(self, loaded_font, font_url)
 
-    return deviation == 0.0
+        if not target_font or target_font == fontface.family:
+            try:
+                handle["font-loaded"][0] = True
+            except Exception as e:
+                raise ValueError("unable to resolve") from e
+
+    RendererHTMLCanvas.load_font_into_web = load_font_into_web_wrapper
+
+
+def compare_func_handle(selenium):
+    @run_in_pyodide(packages=["matplotlib"])
+    def prepare(selenium):
+        from pytest_pyodide.decorator import PyodideHandle
+
+        font_loaded = [False]
+
+        async def compare(ref):
+            import asyncio
+            import io
+
+            import matplotlib.pyplot as plt
+            import numpy as np
+            from PIL import Image
+
+            nonlocal font_loaded
+            while not font_loaded[0]:  # wait until font is loading
+                await asyncio.sleep(0.2)
+
+            canvas_data = plt.gcf().canvas.get_pixel_data()
+            ref_data = np.asarray(Image.open(io.BytesIO(ref)))
+
+            deviation = np.mean(np.abs(canvas_data - ref_data))
+            assert float(deviation) == 0.0
+
+        return PyodideHandle({"compare": compare, "font-loaded": font_loaded})
```

Comment:
How about
```suggestion
        class result:
            compare = compare
            font_loaded = font_loaded
        return PyodideHandle(result)
```
and then elsewhere use `handle.font_loaded` and `handle.compare`.

Reply:
Sounds good, thanks!